src/Expr.h

```diff
@@ -822,29 +822,41 @@ class BitExpr final : public BinaryExpr {
     ExprPtr Reduce(Reducer* c, StmtPtr& red_stmt) override;
 };
 
-class EqExpr final : public BinaryExpr {
+// Intermediary class for comparison operators. Not directly instantiated.
+class CmpExpr : public BinaryExpr {
+protected:
+    CmpExpr(ExprTag tag, ExprPtr op1, ExprPtr op2);
+
+    void Canonicalize() override;
```

Comment:
I'm fine moving them there, though it actually seems more natural to me to have them protected, because `Canonicalize()` is meant for internal use by the object itself, and `WillTransform()` is only meaningful in the context of script optimization.